lib/job-iteration/iteration.rb

```diff
@@ -184,10 +184,11 @@ def iterate_with_enumerator(enumerator, arguments)
         # Deferred until 2.0.0
         # assert_valid_cursor!(index)
 
-        record_unit_of_work do
+        ActiveSupport::Notifications.instrument("each_iteration.iteration", {}) do |tags|
           found_record = true
           each_iteration(object_from_enumerator, *arguments)
           self.cursor_position = index
+          tags.replace(instrumentation_tags)
```

Comment:
Do we really want the updated `cursor_position` here? The notification describes the iteration that just happened, so it makes most sense to me to tag it with the cursor position at the start of that iteration. If we do want to include both, we can give the one after a different name?

Reply:
Okay, that makes sense. I think my confusion comes from the fact that when you enqueue the job with `cursor_position: 1`, it doesn't process id 1, but only 2 and up, so the cursor position and item processed don't quite line up. But this is indeed the `cursor_position` of the thing we just processed, so it makes sense to use this as the tag 👍